addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +219,35 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                this._stopLoaderTimeout();
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                this._stopLoaderTimeout();
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if (!this.thread) {
+                Object.assign(this, { isShowingLoading: false });
+                this._stopLoaderTimeout();
+            } else {
+                const threadCache = this.threadCache;
+                if (threadCache.isLoading) {
+                    if (!this._loaderTimeout) {
+                        this._loaderTimeout = setTimeout(() => {
+                            Object.assign(this, {
+                                isShowingLoading: threadCache.isLoading
+                            });
+                            this._stopLoaderTimeout();
+                        }, 400);
+                    }
+                } else if (this.isShowingLoading) {
+                    Object.assign(this, { isShowingLoading: false });
+                }
```

Comment:
something is off with this branch... don't know if it's because there is no management of the timeout or this code never runs in practice.

Reply:
This code is executed when there is a thread and the threadCache is not loading. It could happen if you create a theadviewer on a thread already loaded. I don't know if it's done somewhere but maybe it's the case ?

It can also be executed if an update is triggered when the threadCache is loaded and the timeout is not yet executed (I mean, he's created and waiting 400 ms, maybe the update is triggered after 200ms). So we set the boolean to false already to not wait the timeout to do it (even if it doesn't change a lot in this situation).

Reply:
We can indeed in this branch cancel the timeout if you think it's best ?

Reply:
Yes, no need to let it run for nothing then, it might even potentially conflict if there is another change in between.

Reply:
Creating a thread viewer on already loaded thread definitely happens, especially with chat windows.
I think the problem I have this this branch is that it shows `isShowingLoading` and its timeout are a bit desynchronized.
Maybe that's because change of thread or thread cache should make changes to both the timeout and recompute `isShowingLoading` at the same time.